# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: xpcom/io/nsLocalFileWin.h
# Commit: 7e19c89e818b
# Full Hash: 7e19c89e818b8cf88661386be394e581c8974ab1
# Author: Bob Owen <bobowencode@gmail.com>
# Date: 2025-10-06 08:04 +0000
# Regressor Bug: 1980886
# File Overlap Count: 1
# Overlapping Files: xpcom/io/nsLocalFileWin.cpp
# Description:
#   Bug 1980886 p3 - Ensure no old inherited ACEs remain when moving files on Windows. r=yjuglaret
# ==============================================================================

diff -r e1b091b14517 -r 7e19c89e818b xpcom/io/nsLocalFileWin.h
--- a/xpcom/io/nsLocalFileWin.h	Mon Oct 06 08:04:25 2025 +0000
+++ b/xpcom/io/nsLocalFileWin.h	Mon Oct 06 08:04:25 2025 +0000
@@ -14,6 +14,7 @@
 #include "nsIFile.h"
 #include "nsILocalFileWin.h"
 #include "nsIClassInfoImpl.h"
+#include "nsWindowsHelpers.h"
 #include "prio.h"
 
 #include "mozilla/Attributes.h"
@@ -52,6 +53,21 @@
   // (com1, com2, etc...) and returns true if so.
   static bool CheckForReservedFileName(const nsString& aFileName);
 
+  /**
+   * Checks whether the inherited ACEs in aChildDacl only come from the parent.
+   *
+   * @param aChildDacl the DACL for the child file object
+   * @param aIsChildDir true if child is a directory
+   * @param aChildSecDesc the SECURITY_DESCRIPTOR for the child, this will
+   *                      correspond to aChildDacl
+   * @param aParentDir the parent nsIFile
+   * @return true if inherited ACEs are only from aParentDir
+   *         false if child has other inherited ACEs or a failure occurs
+   */
+  static bool ChildAclMatchesAclInheritedFromParent(
+      const mozilla::NotNull<ACL*> aChildDacl, bool aIsChildDir,
+      const AutoFreeSecurityDescriptor& aChildSecDesc, nsIFile* aParentDir);
+
   // PRFileInfo64 does not hvae an accessTime field;
   struct FileInfo {
     PRFileType type;